purchase_request_group_by_date/tests/test_purchase_request_group_by.py

```diff
@@ -0,0 +1,183 @@
+
+        # Create UoM
+        self.uom_ten = self.product_uom_model.create(
+            {
+                "name": "Ten",
+                "category_id": self.uom_unit_categ.id,
+                "factor_inv": 10,
+                "uom_type": "bigger",
+            }
+        )
+
+        # Create Supplier
+        self.supplier = self.env["res.partner"].create(
+            {"name": "Supplier", "is_company": True, "company_id": False}
+        )
+
+        # Add supplier to product_1
+        self.product_1.write(
+            {
+                "seller_ids": [
+                    (
+                        0,
+                        0,
+                        {"name": self.supplier.id, "price": 100.0, "company_id": False},
+                    )
+                ]
+            }
+        )
+
+    def procurement_group_run(self, name, origin, product, qty):
+        values = {
+            "date_planned": fields.Datetime.now(),
+            "warehouse_id": self.env.ref("stock.warehouse0"),
+            "route_ids": self.env.ref("purchase_stock.route_warehouse0_buy"),
+            "company_id": self.company_id,
+        }
+        procurements = []
+        procurements.append(
+            self.env["procurement.group"].Procurement(
+                product,
+                qty,
+                product.uom_id,
+                self.location,
+                name,
+                origin,
+                self.company_id,
+                values,
+            )
+        )
+        return self.env["procurement.group"].run(procurements)
+
+    def test_duplicated_purchase_request(self):
+        self.assertEqual(self.env["purchase.request"].search_count([]), 0)
+        has_route = self.procurement_group_run(
+            "Test Purchase Request Single Line",
+            "Test Purchase Request Single Line",
+            self.product_1,
+            4,
+        )
+        self.assertTrue(has_route)
+        self.env["procurement.group"].run_scheduler()
+        pr = self.env["purchase.request"].search(
+            [("origin", "=", "Test Purchase Request Single Line")]
+        )
+        self.assertTrue(pr.to_approve_allowed)
+        self.assertEqual(pr.origin, "Test Purchase Request Single Line")
+
+        prl = self.env["purchase.request.line"].search([("request_id", "=", pr.id)])
+        self.assertEqual(prl.request_id, pr)
+        self.assertEqual(prl.product_qty, 4)
+        self.assertEqual(prl.product_id, self.product_1)
+
+        # another request
+        has_route = self.procurement_group_run(
+            "Test Purchase Request Single Line",
+            "Test Purchase Request Single Line",
+            self.product_1,
+            5,
+        )
+        self.assertTrue(has_route)
+        self.env["procurement.group"].run_scheduler()
+        # make sure no new PR is created
+        self.assertEqual(self.env["purchase.request"].search_count([]), 1)
+        self.assertEqual(len(pr.line_ids), 1)
```

Comment:
```
        # make sure no new PR is created
        self.assertEqual(self.env["purchase.request"].search_count([]), 1)
        import pdb;pdb.set_trace();
>       self.assertEqual(len(pr.line_ids), 1)
E       AssertionError: 2 != 1

odoo/external-src/purchase-workflow/purchase_request_group_by_date/tests/test_purchase_request_group_by.py:113: AssertionError

```

Reply:
afais in model:
```
(Pdb) self.company_group_purchase_request
False
```

so parameter isn't properly set in test case, that's why mechanism doesnt work.

Reply:
@simahawk  if you would have some time, can you take a look here? Issue is that i am moving what we did recently for gzt to another oca module where it can be enabled/diabled by parameter. But test fails due to assertion which is caused by fact that travis doesnt see flag set on the company (i tested manually in UI and it seems to works correctly). 
I am not sure if it is enough to do in test:     `self.company_id.group_purchase_request = True ` like it is in l.28 or i should simulate it with some res.config records?